### PR TITLE
feat: enable change assistant name and avatar

### LINKED ISSUE #2894: feat: User can change Assistant name and avatar
# Problem
We got requests from many users to make the assistant field editable:
Screenshot 2024-05-13 at 12 13 59 PM
# Design
https://www.figma.com/design/DYfpMhf8qiSReKvYooBgDV/Jan-App-(version-2)?node-id=3896-57872&t=dKPqEv1WkJxHKSeY-4
- Allow users to change assistant avatar
- Allow users to change assistant name

Screenshot 2024-05-15 at 1 45 09 AM

### PR DESCRIPTION
**Pull request by urmauur**
## Describe Your Changes

- [X] Enable user change assistant name and avatar
- [X] Conditional if no avatar will fallback using default Jan icon
- [X] No required size image, once user upload no square image, image will show as a cover and center position

Screenshot 2024-09-05 at 14 02 15
Screenshot 2024-09-05 at 14 02 09
Screenshot 2024-09-05 at 14 02 00

## Fixes Issues

- Closes #2894 
- Closes #

## Self Checklist

- [ ] Added relevant comments, esp in complex areas
- [ ] Updated docs (for bug fixes / features)
- [ ] Created issues for follow-up changes or refactoring needed

### CODE REVIEW

**Comment by dan-homebrew** [comment]
@urmauur Can we add a couple of test cases for this: 

- Default case: e.g. Jan
- User-specified case: test that the changes were persisted

Is there a way we can do this as a pure JS backend test (UI testing seems too heavy for this)

File: core/src/types/assistant/assistantEntity.ts [hunk 1]
```diff
@@ -12,7 +12,7 @@ export type AssistantTool = {
 
 export type Assistant = {
   /** Represents the avatar of the user. */
-  avatar: string
+  avatar?: string
```
Comment (dan-homebrew) [inline core/src/types/assistant/assistantEntity.ts:15]: Test coverage requested

File: web/screens/Thread/ThreadRightPanel/index.tsx [hunk 2]
```diff
@@ -150,6 +155,44 @@ const ThreadRightPanel = () => {
...
+    [activeThread, updateThreadMetadata]
+  )
+
+  const onAssistantAvatarChanged = useCallback(
+    (e: React.ChangeEvent<HTMLInputElement>) => {
+      if (activeThread && e.target.files && e.target.files[0]) {
+        const file = e.target.files[0]
+        const reader = new FileReader()
+        reader.onloadend = () => {
+          updateThreadMetadata({
```
Comment (dan-homebrew) [inline web/screens/Thread/ThreadRightPanel/index.tsx:180]: Do we only update the Assistant for the Thread, or for overall Jan?

**Comment by dan-homebrew** [comment]
@urmauur Additionally, when user specifies an Avatar and Assistant name, what is our behavior? 

- Persist for thread?
- Call Assistant update API? (do we have this yet?)

cc @louis-jan

**Review by dan-homebrew — CHANGES_REQUESTED** [review]
Test coverage requested

File: web/screens/Thread/ThreadRightPanel/index.tsx (hunk 2 above)
Comment (urmauur) [reply web/screens/Thread/ThreadRightPanel/index.tsx:180]: currently we update on thread level, bc assistant only for template when user request create thread @dan-homebrew

**Comment by dan-homebrew** [comment]
@urmauur @louis-jan I would like to pause this PR for the next 2 days, in favor of correctly dealing with this as part of an Assistants epic: 

- Jan is the default Assistant
- User can customize the Assistant (i.e. avatar, name)
- User can customize the System Prompt (i.e. Assistant prompt)
- Assistants are persisted across threads

This will call the Assistants API (e.g. PUT `/assistants/jan`), which we can build as part of Jan's 0.5.5 release in the later half of this sprint. 

This will set the correct API foundation to allow Jan (the app) to support multiple Assistants in the future. 

I am not in favor of piecemeal fixes - let's fix this at the Assistants API level.

**Comment by urmauur** [comment]
Closed PR https://github.com/janhq/jan/pull/3560#issuecomment-2337350882

**Pull request closed without merging**